src/helpers/gateway/send.rs

```diff
@@ -36,18 +36,26 @@ pub(super) struct GatewaySender {
     channel_id: ChannelId,
     ordering_tx: OrderingSender,
     total_records: TotalRecords,
+
+    message_size: usize,
+    pending_records: Mutex<HashSet<usize>>,
```

Comment:
you seem to track indices inside the ordering sender, do you still need this?

Reply:
removed.

Reply:
Removed